libbeat/cfgfile/list.go

```diff
@@ -85,7 +85,7 @@ func (r *RunnerList) Reload(configs []*reload.ConfigWithMeta) error {
 	for hash, runner := range stopList {
 		r.logger.Debugf("Stopping runner: %s", runner)
 		delete(r.runners, hash)
-		go runner.Stop()
+		runner.Stop()
```

Comment:
I'm curious if this would make `Reload()` significantly slower when there are multiple runners to stop. It seems like it would be closer to the original behaviour use a WaitGroup to begin stopping all the runners concurrently.

Reply:
It looks like it was originally synchronous https://github.com/elastic/beats/pull/7172#discussion_r190881918, and from the linked comment for exactly the reason you are making it synchronous again.

Reply:
I get your concern, but I don't think we have an option here. On my tests the inputs being stopped are always the same as the ones being started again, so not waiting does lead to inconsistencies. It also makes it impossible to track which filestream IDs are being used (the "new" input is started before the old stops and we get the error log about data duplication).

I can use a WaitGroup so at least the runners stop concurrently, which should reduce the performance impact.